File: packages/iroh-http-tauri/android/src/main/java/com/iroh/http/discovery/IrohDiscoveryPlugin.kt
package com.iroh.http.discovery

/**
 * Stub Android native plugin for iroh-http mDNS discovery.
 * Full implementation uses Android's NsdManager for local peer discovery.
 */
class IrohDiscoveryPlugin {
    fun startDiscovery(serviceName: String, advertise: Boolean) {
        // TODO: implement using NsdManager
    }

    fun stopDiscovery() {
        // TODO: stop NsdManager
    }
}

File: packages/iroh-http-tauri/ios/Sources/IrohDiscoveryPlugin.swift
import Foundation

/// Stub iOS native plugin for iroh-http mDNS discovery.
/// Full implementation uses Network.framework NWBrowser for local peer discovery.
@objc public class IrohDiscoveryPlugin: NSObject {
    @objc public static func startDiscovery(serviceName: String, advertise: Bool) {
        // TODO: implement using NWBrowser / NWListener
    }

    @objc public static func stopDiscovery() {
        // TODO: stop NWBrowser / NWListener
    }
}
